modules/repofiles/update.go

```diff
@@ -495,9 +495,18 @@ func PushUpdate(repo *models.Repository, branch string, opts PushUpdateOptions)
 		return err
 	}
 
-	log.Trace("TriggerTask '%s/%s' by %s", repo.Name, branch, pusher.Name)
+	if opts.NewCommitID != git.EmptySHA {
+		if err = models.RemoveDeletedBranch(repo.ID, opts.Branch); err != nil {
```

Comment:
Delete branch on normal push? 🤔 Maybe this goes in the `else` part.

Reply:
This will remove the deleted branch information from database, it's not to delete the branch.
The table deleted branch is designed to restore branch deletion.

Reply:
Ah! thank you that's a remarkably badly named function.